source/common/json/json_loader.cc

```diff
@@ -1,178 +1,194 @@
 #include "json_loader.h"
 
-// Do not let jansson leak outside of this file.
+// Do not let RapidJson leak outside of this file.
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wold-style-cast"
-#include "jansson.h"
+#include "rapidjson/document.h"
+#include "rapidjson/error/en.h"
+#include "rapidjson/istreamwrapper.h"
 #pragma GCC diagnostic pop
 
 namespace Json {
 
-FileLoader::FileLoader(const std::string& file_path) {
-  json_error_t error;
-  json_ = json_load_file(file_path.c_str(), 0, &error);
-  if (!json_) {
-    throw Exception(fmt::format("json parsing error: {} line: {} column: {}", error.text,
-                                error.line, error.column));
+/**
+ * Implementation of AbstractObject.
+ */
+class AbstractObjectImpl : public AbstractObject {
+public:
+  AbstractObjectImpl(const rapidjson::Document& document, const std::string& name) : name_(name) {
+    document_.CopyFrom(document, document_.GetAllocator());
```

Comment:
We should not be copying anything. If needed a reference to the parent document should be used.